xarray/test/test_dataset.py

```diff
@@ -1658,6 +1658,59 @@ def test_resample_and_first(self):
             actual = ds.resample('3H', 'time', how=how)
             self.assertDatasetEqual(expected, actual)
 
+    def test_resample_by_mean_with_keep_attrs(self):
```

Comment:
I have the same thoughts on these methods as the other -- you can consolidate them into a single test with a loop over `how`